### PR TITLE
[Flexidag] set flexidag effective height when executing genesis

### PR DESCRIPTION
**Pull request by simonjiao**
1. update GenesisConfig
2. fix genesis-config tests
3. update force-upgrade init_script function verification

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

### CODE REVIEW

**Comment by nkysg** [comment]
related starcoin-framework https://github.com/starcoinorg/starcoin-framework/commits/dag-master/

File: Cargo.toml [hunk 1]
```diff
@@ -451,7 +451,7 @@ starcoin-crypto = { git = "https://github.com/starcoinorg/starcoin-crypto", rev
 starcoin-decrypt = { path = "commons/decrypt" }
 starcoin-dev = { path = "vm/dev" }
 starcoin-executor = { path = "executor" }
-starcoin-framework = { git = "https://github.com/starcoinorg/starcoin-framework", rev = "76f3436af2494fc332f14dbacb3d7847cec711c0" }
+starcoin-framework = { git = "https://github.com/starcoinorg/starcoin-framework", rev = "94bcd77e80232b169cf95754ef4e87775645afcd" }
```
Comment (nkysg) [inline Cargo.toml:454]: related starcoin-framework https://github.com/starcoinorg/starcoin-framework/commits/dag-master/

**Review by sanlee42 — CHANGES_REQUESTED** [review]

File: chain/src/chain.rs [hunk 2]
```diff
@@ -1260,15 +1261,22 @@ impl BlockChain {
...
+        #[cfg(not(feature = "force-deploy"))]
+        let valid_txn_num = {
+            // Silence unused variable warning
+            let _ = chain_id;
+            vec_transaction_info.len() == transactions.len()
+        };
+
         verify_block!(
             VerifyBlockField::State,
-            {
```
[the remark this answers is not in the document]
Comment (simonjiao) [reply chain/src/chain.rs:1265]: hmm, need to update check logic when stdlib version is higher than 11.

File: config/src/genesis_config.rs [hunk 3]
```diff
@@ -827,7 +832,7 @@ pub static G_HALLEY_BOOT_NODES: Lazy<Vec<MultiaddrWithPeerId>> = Lazy::new(|| {
 
 pub static G_HALLEY_CONFIG: Lazy<GenesisConfig> = Lazy::new(|| {
     // let stdlib_version = StdlibVersion::Latest;
-    let stdlib_version = StdlibVersion::Version(11);
+    let stdlib_version = StdlibVersion::Version(12);
```
Comment (nkysg) [inline config/src/genesis_config.rs:835]: We use version 12 , not use Latest?

**Review by sanlee42 — APPROVED** [review]

File: config/src/genesis_config.rs (hunk 3 above)
Comment (simonjiao) [reply config/src/genesis_config.rs:835]: Currently the latest stdlib version is 12.

**Review by nkysg — APPROVED** [review]

**Pull request merged**